components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -862,15 +865,31 @@ public AccessTokenDO getAccessToken(String accessTokenIdentifier, boolean includ
 
             if (includeExpired) {
                 if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled()) {
-                    sql = SQLQueries.RETRIEVE_ACTIVE_EXPIRED_ACCESS_TOKEN_IDP_NAME;
+                    if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled() && tenantDomain != null) {
```

Comment:
Let's keep a boolean at the class level and use that. Calling the Util method multiple times can be expensive.

Reply:
Resolved